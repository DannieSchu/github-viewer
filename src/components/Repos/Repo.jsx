import React from 'react';
import PropTypes from 'prop-types';

const Repo = ({ name, html_url, language, created_at, updated_at }) => {
  return (
    <>
      <a href={html_url} target="blank">{name}</a>
      <p>Language: {language}</p>
      <p>Created: {created_at}</p>
      <p>Updated: {updated_at}</p>
    </>
  );
};

Repo.propTypes = {
  name: PropTypes.string.isRequired,
  html_url: PropTypes.string.isRequired,
  language: PropTypes.string.isRequired,
  created_at: PropTypes.string.isRequired,
  updated_at: PropTypes.string.isRequired
};

export default Repo;
