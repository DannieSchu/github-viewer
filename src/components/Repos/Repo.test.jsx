import React from 'react';
import { shallow } from 'enzyme';
import Repo from './Repo.jsx';

describe('Repo', () => {
  it('matches a snapshot', () => {
    const wrapper = shallow(
      <Repo 
        name="About-Me"
        html_url="https://github.com/DannieSchu/About-Me"
        language="JavaScript"
        created_at="2020-01-14T00:49:15Z"
        updated_at="2020-01-14T00:49:15Z"
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
