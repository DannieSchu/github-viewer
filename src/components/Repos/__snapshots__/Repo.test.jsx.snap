// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo matches a snapshot 1`] = `
<Fragment>
  <a
    href="https://github.com/DannieSchu/About-Me"
    target="blank"
  >
    About-Me
  </a>
  <p>
    Language: 
    JavaScript
  </p>
  <p>
    Created: 
    2020-01-14T00:49:15Z
  </p>
  <p>
    Updated: 
    2020-01-14T00:49:15Z
  </p>
</Fragment>
`;
